sdk/rust/simple-searcher/src/main.rs

```diff
@@ -0,0 +1,220 @@
+            args.private_key_svm.clone()
+        }
+    };
+
+    match private_key {
+        Some(private_key) => {
+            if let Err(e) = submit_opportunity(ws_client, opportunity, private_key).await {
+                eprintln!("Failed to submit opportunity: {:?}", e);
+            }
+        }
+        None => {
+            eprintln!("Private key not provided");
+        }
+    }
+}
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    let args: RunOptions = RunOptions::parse();
+    if let Some(private_key) = args.private_key_evm.clone() {
+        private_key
+            .parse::<LocalWallet>()
+            .map_err(|e| anyhow!("Invalid evm private key: {}", e))?;
+    }
+
+    let client = Client::try_new(ClientConfig {
+        http_url: args.server_url.clone(),
+        api_key:  args.api_key.clone(),
+    })
+    .map_err(|e| {
+        eprintln!("Failed to create client: {:?}", e);
+        anyhow!("Failed to create client")
+    })?;
+
+    let ws_client = client.connect_websocket().await.map_err(|e| {
+        eprintln!("Failed to connect websocket: {:?}", e);
+        anyhow!("Failed to connect websocket")
+    })?;
+
+    let opportunities = client
+        .get_opportunities(Some(GetOpportunitiesQueryParams {
+            chain_id:       Some(args.chains[0].clone()),
+            mode:           OpportunityMode::Live,
+            permission_key: None,
+            limit:          100,
+            from_time:      Some(OffsetDateTime::now_utc() - Duration::days(1)),
+        }))
+        .await
+        .map_err(|e| {
+            eprintln!("Failed to get opportunities: {:?}", e);
+            anyhow!("Failed to get opportunities")
+        })?;
+
+    opportunities.iter().for_each(|opportunity| {
+        tokio::spawn(handle_opportunity(
+            ws_client.clone(),
+            opportunity.clone(),
+            args.clone(),
+        ));
+    });
+
+    ws_client
+        .chain_subscribe(args.chains.clone())
+        .await
+        .map_err(|e| {
+            eprintln!("Failed to subscribe chains: {:?}", e);
+            anyhow!("Failed to subscribe chains")
+        })?;
+
+    let mut stream = ws_client.get_update_stream();
+    let mut block_hash_map = HashMap::new();
+    while let Some(update) = stream.next().await {
+        let update = match update {
+            Ok(update) => update,
+            Err(e) => {
+                eprintln!("The stream is fallen behind: {:?}", e);
+                continue;
+            }
+        };
+
+        match update {
+            ServerUpdateResponse::NewOpportunity { opportunity } => {
+                tokio::spawn(handle_opportunity(
+                    ws_client.clone(),
+                    opportunity,
+                    args.clone(),
+                ));
+            }
+            ServerUpdateResponse::SvmChainUpdate { update } => {
+                block_hash_map.insert(update.chain_id.clone(), update.blockhash);
+                println!("SVM chain");
```

Comment:
maybe add some more info here

Reply:
This part will be refactored when we add the svm part